app/views/course/assessment/answer/programming/_test_cases.html.slim

```diff
@@ -1,127 +1,79 @@
-- test_cases = question.test_cases
-- public_test_cases = test_cases.select(&:public_test?)
-- private_test_cases = test_cases.select(&:private_test?)
-- evaluation_test_cases = test_cases.select(&:evaluation_test?)
-- submission = answer.submission
 - auto_grading = answer.auto_grading ? answer.auto_grading.actable : nil
 - answer_test_results_hash = auto_grading ? \
     auto_grading.test_results.map { |result| [result.test_case, result] }.to_h : \
     {}
+- is_grader = can?(:grade, answer.answer)
+/ Manually list test case types so they're in the desired display order
+- test_case_types = ['public_test', 'private_test', 'evaluation_test']
+- grader_test_case_types = ['private_test', 'evaluation_test']
```

Comment:
As suggested, will solve this by moving the test case table display code into a separate partial.